packages/webpack-config/webpack/utils/getPaths.js

```diff
@@ -2,25 +2,44 @@ const path = require('path');
 const findWorkspaceRoot = require('find-yarn-workspace-root');
 const fs = require('fs');
 const ConfigUtils = require('@expo/config');
-const possibleMainFiles = [
-  'index.web.ts',
-  'index.ts',
-  'index.web.tsx',
-  'index.tsx',
-  'src/index.web.ts',
-  'src/index.ts',
-  'src/index.web.tsx',
-  'src/index.tsx',
-  'index.web.js',
-  'index.js',
-  'index.web.jsx',
-  'index.jsx',
-  'src/index.web.js',
-  'src/index.js',
-  'src/index.web.jsx',
-  'src/index.jsx',
+
+const possibleMainFiles = ['index', 'src/index'];
+
+const extensions = [
+  'web.mjs',
+  'mjs',
+  'web.js',
+  'js',
+  'web.ts',
+  'ts',
+  'web.tsx',
+  'tsx',
+  'json',
+  'web.jsx',
+  'jsx',
 ];
 
+const resolveModuleExists = (resolveFn, filePath) => {
+  const extension = extensions.find(extension =>
+    fs.existsSync(resolveFn(`${filePath}.${extension}`))
```

Comment:
sometimes people already include extensions in their module names e.g. `require('./x.js')`